.github/workflows/test_packages.yml

```diff
@@ -38,54 +38,62 @@ jobs:
-      amdgpu_families: ${{ inputs.amdgpu_families }}
-      test_runs_on: ${{ inputs.test_runs_on }}
-      platform: ${{ inputs.platform }}
+      - name: "Configuring CI options"
+        env:
+          PLATFORM: ${{ inputs.platform }}
+          THEROCK_BIN_DIR: "./build/bin"
+        id: configure
+        run: python ./build_tools/github_actions/fetch_test_configurations.py
 
-  # PRIM tests
-  test_rocprim:
-    name: 'Test rocPRIM math-lib'
-    uses: './.github/workflows/test_rocprim.yml'
-    needs: test_sanity_check
-    with:
-      artifact_run_id: ${{ inputs.artifact_run_id }}
-      amdgpu_families: ${{ inputs.amdgpu_families }}
-      test_runs_on: ${{ inputs.test_runs_on }}
-      platform: ${{ inputs.platform }}
+  test_components:
+    name: 'Test ${{ matrix.components.name }}'
+    runs-on: ${{ inputs.test_runs_on }}
+    needs: configure_test_matrix
+    # skip tests if no test matrix to run
+    if: ${{ needs.configure_test_matrix.outputs.components != '[]' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        components: ${{ fromJSON(needs.configure_test_matrix.outputs.components) }}
+    defaults:
+      run:
+        shell: bash
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+      ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
+      OUTPUT_ARTIFACTS_DIR: ${{ github.workspace }}/build
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: "ROCm/TheRock"
 
-  test_rocthrust:
-    name: 'Test rocTHRUST math-lib'
-    uses: './.github/workflows/test_rocthrust.yml'
-    needs: test_sanity_check
-    with:
-      artifact_run_id: ${{ inputs.artifact_run_id }}
-      amdgpu_families: ${{ inputs.amdgpu_families }}
-      test_runs_on: ${{ inputs.test_runs_on }}
-      platform: ${{ inputs.platform }}
+      - name: Run setup test environment workflow
+        uses: './.github/actions/setup_test_environment'
+        with:
+          ARTIFACT_RUN_ID: ${{ env.ARTIFACT_RUN_ID }}
+          AMDGPU_FAMILIES: ${{ inputs.amdgpu_families }}
+          OUTPUT_ARTIFACTS_DIR: ${{ env.OUTPUT_ARTIFACTS_DIR }}
+          VENV_DIR: ${{ env.VENV_DIR }}
+          FETCH_ARTIFACT_ARGS: ${{ matrix.components.artifact_flags }}
+          PLATFORM: ${{ inputs.platform }}
 
-  test_hipcub:
-    name: 'Test hipCUB math-lib'
-    uses: './.github/workflows/test_hipcub.yml'
-    needs: test_sanity_check
-    with:
-      artifact_run_id: ${{ inputs.artifact_run_id }}
-      amdgpu_families: ${{ inputs.amdgpu_families }}
-      test_runs_on: ${{ inputs.test_runs_on }}
-      platform: ${{ inputs.platform }}
+      - name: Install additional packages
+        if: ${{ inputs.platform == 'linux' && (matrix.components.name == 'rocblas' || matrix.components.name == 'hipblaslt') }}
+        run: sudo apt install libgfortran5 -y
```

Comment:
This could be moved into the matrix also, with each project having both a `test_command` and a `setup_command` (that could be empty).

We could consider something like what @lamikr has been setting up in https://github.com/ROCm/TheRock/blob/main/experimental/rockbuilder/projects/pytorch_audio.cfg, with config (or python) files that give each project a place to put its own versions of steps per phase (init, cleanup, build, test, postprocess, etc.)